contentcuration/contentcuration/tests/utils/test_exercise_creation.py

```diff
@@ -1122,5 +1128,697 @@ def test_image_with_zero_width(self):
+
+        self.assertEqual(exercise_file.checksum, "51ba0d6e3c7f30239265c5294abe6ac5")
+
+    def test_question_with_mathematical_content(self):
+        """Test QTI generation for questions containing mathematical formulas converted to MathML"""
+        assessment_id = "dddddddddddddddddddddddddddddddd"
+        item = self._create_assessment_item(
+            exercises.SINGLE_SELECTION,
+            "Solve the equation $$\\frac{x}{2} = 3$$ for x. What is the value of x?",
+            [
+                {"answer": "6", "correct": True, "order": 1},
+                {"answer": "3", "correct": False, "order": 2},
+                {"answer": "1.5", "correct": False, "order": 3},
+                {"answer": "9", "correct": False, "order": 4},
+            ],
+            assessment_id=assessment_id,
+        )
+
+        exercise_data = {
+            "mastery_model": exercises.M_OF_N,
+            "randomize": True,
+            "n": 1,
+            "m": 1,
+            "all_assessment_items": [item.assessment_id],
+            "assessment_mapping": {item.assessment_id: exercises.SINGLE_SELECTION},
+        }
+
+        self._create_qti_zip(exercise_data)
+        exercise_file = self.exercise_node.files.get(preset_id=format_presets.QTI_ZIP)
+        zip_file = self._validate_qti_zip_structure(exercise_file)
+
+        qti_id = hex_to_qti_id(assessment_id)
+
+        # Check the QTI XML for mathematical content conversion to MathML
+        expected_item_file = f"items/{qti_id}.xml"
+        actual_item_xml = zip_file.read(expected_item_file).decode("utf-8")
+
+        # Expected QTI item XML content with MathML conversion
+        expected_item_xml = f"""<?xml version="1.0" encoding="UTF-8"?>
+    <qti-assessment-item xmlns="http://www.imsglobal.org/xsd/imsqtiasi_v3p0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.imsglobal.org/xsd/imsqtiasi_v3p0 https://purl.imsglobal.org/spec/qti/v3p0/schema/xsd/imsqti_asiv3p0p1_v1p0.xsd" identifier="{qti_id}" title="Test QTI Exercise 1" adaptive="false" time-dependent="false" language="en-US" tool-name="kolibri" tool-version="0.1">
+        <qti-response-declaration identifier="RESPONSE" cardinality="single" base-type="identifier">
+            <qti-correct-response>
+                <qti-value>choice_0</qti-value>
+            </qti-correct-response>
+        </qti-response-declaration>
+        <qti-outcome-declaration identifier="SCORE" cardinality="single" base-type="float" />
+        <qti-item-body>
+            <qti-choice-interaction response-identifier="RESPONSE" shuffle="true" max-choices="1" min-choices="0" orientation="vertical">
+                <qti-prompt>
+                    <p>Solve the equation <math display="inline"><semantics><mrow><mfrac><mrow><mi>x</mi></mrow><mrow><mn>2</mn></mrow></mfrac><mo>=</mo><mn>3</mn></mrow><annotation encoding="application/x-tex">\\frac{{x}}{{2}} = 3</annotation></semantics></math> for x. What is the value of x?</p>
```

Comment:
This xml seemed to cause issues for the QTI Sandbox - possibly an issue with processing the XML with MathML/annotation included.